### DISCUSSION TITLE
Method to mantain latest versions automatically

### DISCUSSION THREAD

**Discussion by piraces** [microsoft/winget-pkgs#7503] [Ideas] [answered]
Hello,

I was wondering if there is or we could develop a method to mantain the latest versions of packages automatically.
For example, taking a look at the tool [Chocolatey](https://chocolatey.org/), they have [community templates](https://github.com/chocolatey-community/chocolatey-packages-template) and [tools](https://github.com/majkinetor/au/blob/master/README.md) to automatically update their packages to the latest verions.

Do we have or it is planned to have a think like this for maintainers? I think it should be a great idea and feature to be up-to-date for sure.

I mention Chocolatey but is not my intention to compare this winget with other tools.

**Comment by jedieaston** [comment, marked as answer]
It's already happening for many packages, see @wingetbot's contributions in the repo. More tooling is coming, it's being tracked in #5833 and #5418 (as well as many other places, I'm sure).

**Reply by piraces** [reply]
Awesome. Thank you!

**Comment by denelon** [comment]
We regularly scan "all" of the manifests. If we detect a hash mismatch, then we try to download the package and generate a new one. Then a PR is submitted to correct it. If that passes, then it gets updated in the repository.

We're working on code to identify manifests where the download fails. If there is a "working" version of the package, then we delete the manifest with the non-functional package URL. If there is no working version available, then we will automatically create an Issue like https://github.com/microsoft/winget-pkgs/issues/7899 to let the community know we could no longer identify a valid URL. The code is being tested currently, and we're hoping to roll it out in the next week or two.
